WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java

```diff
@@ -359,33 +359,32 @@ public static boolean isFollowing(
         }
     }
 
-    public static ReaderTag getValidTagForSharedPrefs(
-            ReaderTag tag,
+    public static @NonNull ReaderTag getValidTagForSharedPrefs(
+            @NonNull ReaderTag tag,
```

Comment:
The name `tag` here is not very descriptive. Even after working on this code I am not entirely sure what it translates to. The closest one I can think of it is a `defaultTag`, but that is already referring to something else in the codebase, so using a more descriptive name here would be very helpful to understand the code.

Alternatively, the signature of the method could be changed to make it not depend on this. The method does a lot of things at the same time so splitting it up into more meaningful units of work would be an even better option.

Reply:
Currently the `tag` parameter is basically the `currentTag`; maybe that can make sense? Let me know wdyt 😊

Reply:
While naming parameters I try to make them meaningful in the function's own context. Although `currentTag` is probably meaningful where this method is called, it is not clear to me why it's the one that's selected as return value in many of these cases.

I think I'd prefer to re-write this method to return `@Nullable` and not take `ReaderTag` as a parameter so the caller is the one that's deciding what to do when a valid tag is not found. If it was Kotlin, I'd go one more step and utilize a `sealed class` here so the meaning is even more clear. However, since this is all legacy code and most likely see a full refactor some time soon, it's probably not worth spending much time on it. I just make it a point to share my thoughts about these things as a good discussion topic and if agreed, something to apply in the future PRs.

Since `currentTag` and `tag` doesn't make much of a difference here, I think I prefer to merge the PR as is just to save some time. I am happy to make the change if you feel otherwise.

Reply:
All this comment makes sense. Also agree that `ReaderPostListFragment` has grown over time and we could benefit of some refactoring. Probably this one could be a good candidate after the `EditPostActivity` that has been already tackled.